lib/commands/smapi/cli-customization-processor.js

```diff
@@ -128,31 +135,39 @@ class CliCustomizationProcessor {
             }
             const json = this._shouldParseAsJson(property, definitions);
             const bodyPath = `${parentName}${BODY_PATH_DELIMITER}${key}`;
+            const required = this._isRequired(secondLevelDefinition, key, parentRequired);
             customizedParameter = { name: `${parentName} ${key}`, description, rootName, required, bodyPath, enum: enumOptions, json, type };
             this._addCustomizedParameter(customizationMetadata, customizedParameter);
         });
         if (secondLevelDefinition.enum) {
             const { type } = secondLevelDefinition;
             const { description, enumOptions } = this._mapEnum(parentDescription, secondLevelDefinition);
-            customizedParameter = { name: parentName, description, rootName, required, bodyPath: parentName, enum: enumOptions, type };
+            customizedParameter = { name: parentName,
+                description,
+                rootName,
+                required: parentRequired,
+                bodyPath: parentName,
+                enum: enumOptions,
+                type };
```

Comment:
minor style change suggestion, curly brace starts a new line and ends in a new line

Reply:
How about https://eslint.org/docs/rules/object-curly-newline adding this rule for the codebase later, and with a minProperties set as 3?